Makefile

```diff
@@ -1,10 +1,29 @@
 .DEFAULT_GOAL := help
 
-PKGS := $(shell go list ./... | grep -v /vendor/)
-TAG   = $(shell git describe --always --tags --abbrev=0 HEAD)
-LAST  = $(shell git describe --always --tags --abbrev=0 HEAD^)
-BODY  = "`git log ${LAST}..HEAD --oneline --decorate` `printf '\n\#\#\# [Build Info](${BUILD_URL})'`"
-DATE  = $(shell date +'%d%m%y')
+PKGS   := $(shell go list ./... | grep -v /vendor/)
+TAG    := $(shell git describe --always --tags --abbrev=0 HEAD)
+LAST   := $(shell git describe --always --tags --abbrev=0 HEAD^)
+BODY   := "`git log ${LAST}..HEAD --oneline --decorate` `printf '\n\#\#\# [Build Info](${BUILD_URL})'`"
+DATE   := $(shell date +'%d%m%y')
+
+# Ensure we have an unambiguous GOPATH.
+GOPATH := $(shell go env GOPATH)
+
+ifneq "$(or $(findstring :,$(GOPATH)),$(findstring ;,$(GOPATH)))" ""
+  $(error GOPATHs with multiple entries are not supported)
+endif
+
+GOPATH := $(realpath $(GOPATH))
+ifeq ($(strip $(GOPATH)),)
+  $(error GOPATH is not set and could not be automatically determined)
+endif
+
+SRCDIR := $(GOPATH)/src/
+
+ifeq ($(filter $(GOPATH)%,$(CURDIR)),)
+  GOPATH := $(shell mktemp -d "/tmp/dep.XXXXXXXX")
```

Comment:
now that the source code is checkout under GOPATH, maybe you don't need to change GOPATH into /tmp anymore?

Reply:
This is about what I wrote on #91 we can drop this and make the makefile simpler but that will force  people that want to contribute to clone this into a `GOPATH/src` folder. With this implementation the temp folder is only used if the project is cloned outside of `GOPATH`.